dataset/src/main/scala/frameless/ops/AggregateTypes.scala

```diff
@@ -0,0 +1,19 @@
+package frameless
+package ops
+
+import shapeless._
+
+trait AggregateTypes[V, U <: HList] {
```

Comment:
AFAIK `Comapped.Aux[TC, TypedAggregate[T, ?], C]` desugars to
`Lambda[X => Comapped.Aux[TC, TypedAggregate[T, X], C]]`, that might be the source of the problem.